src/Features/Core/Portable/EditAndContinue/EditAndContinueWorkspaceService.cs

```diff
@@ -418,30 +408,82 @@ public async Task<(SolutionUpdateStatus Summary, ImmutableArray<Deltas> Deltas)>
         public void CommitSolutionUpdate()
         {
             var editSession = _editSession;
-
             Contract.ThrowIfNull(editSession);
 
-            var pendingUpdate = Interlocked.Exchange(ref _pendingUpdate, null);
-            Contract.ThrowIfNull(pendingUpdate);
-
+            var pendingUpdate = editSession.RetrievePendingUpdate();
             editSession.DebuggingSession.CommitSolutionUpdate(pendingUpdate);
             editSession.ChangesApplied();
         }
 
         public void DiscardSolutionUpdate()
         {
-            Contract.ThrowIfNull(_editSession);
-
-            var pendingUpdate = Interlocked.Exchange(ref _pendingUpdate, null);
-            Contract.ThrowIfNull(pendingUpdate);
+            var editSession = _editSession;
+            Contract.ThrowIfNull(editSession);
 
+            var pendingUpdate = editSession.RetrievePendingUpdate();
             foreach (var moduleReader in pendingUpdate.ModuleReaders)
             {
                 moduleReader.Dispose();
             }
         }
 
-        public async Task<LinePositionSpan?> GetCurrentActiveStatementPositionAsync(ActiveInstructionId instructionId, CancellationToken cancellationToken)
+        public async Task<ImmutableArray<ImmutableArray<(LinePositionSpan, ActiveStatementFlags)>>> GetBaseActiveStatementSpansAsync(ImmutableArray<DocumentId> documentIds, CancellationToken cancellationToken)
+        {
+            var editSession = _editSession;
+            if (editSession == null)
+            {
+                return default;
+            }
+
+            var lastCommittedSolution = editSession.DebuggingSession.LastCommittedSolution;
```

Comment:
questoin, what counts as 'committed'?  when the user says "run/step" and edits are applied?

Reply:
Yes. Committed means it matches the state of the debuggee.